### ISSUE TITLE
showOpenDialog() unexpected behaviors

### ISSUE THREAD

**Issue by chrilith** [electron/electron#12737]
* Electron version: 2.0.0-beta.8 and lower
* Operating system: macOS

### Expected behavior

Using `filters` option, all filters should properly appear, showing defined names and extensions in a dropdown list.

### Actual behavior

There is a pending #11959 which may fix this problem.
Problem: when using a wildcard filter, other filters don't appear.

Another problem is that after trying to quit (Cmd+Q) when the dialog is open it will then be impossible to quit the app without the "force quit" option.

### How to reproduce

For instance, try:
```javascript
	filters: [
		{ name: "All Files", extensions: ["*"] },
		{ name: "Interchange File Format", extensions: ["lbm", "iff", "bl1", "bl2", "bl3"] },
		{ name: "Atari Degas Elite", extensions: ["pi1", "pi2", "pi3", "pc1", "pc2", "pc3"] }
	]
```

Nothing will appear in the dialog. Now remove the first filter and the filters will appear but still showing extensions only as stated in #12571.

Thanks!

**Comment by codebytere** [comment]
I can take a look at this early next week!

**Comment by chrilith** [comment]
Great thanks!

**Comment by AdrianoFerrari** [comment]
I am experiencing another more serious issue, that's possibly related.

If I have only one non-wildcard filter, clicking on it causes the entire application to crash.
Here's a screen recording of this:
https://www.dropbox.com/s/uc2zq4bj0bfg4g7/dialog-format-bug-2018-04-28_13.06.45.mkv?dl=0

(might not be a factor, but also note that `gko` is a custom extension of mine, that I didn't register with macOS on install)

**Comment by codebytere** [comment]
@AdrianoFerrari @chrilith this will potentially be solved with https://github.com/electron/electron/pull/12779; it was merged but not backported so it didn't go out with the latest betas. 

@AdrianoFerrari, can you please provide me with the filters/code you used to facilitate the crash in the video above? Thanks 😁

**Comment by AdrianoFerrari** [comment]
Sure will do, within next 45min

**Comment by AdrianoFerrari** [comment]
Here's the code:
https://github.com/gingko/client/commit/a402d9dd6ce1c9151f044a70d237f79c170b2b53

And the resulting build with bug
https://github.com/gingko/client/releases/download/untagged-9aa6ec48579e27d0830b/gingko-client-2.0.12-mac.zip

**Comment by AdrianoFerrari** [comment]
Yes, seems to be fixed in 2.0.1.
Thanks!

**Comment by chrilith** [comment]
Confirmed too! Appart from the dropdown list that doesn't fit its content everything seems to work fine.

Thanks
